process/block/shardblock.go

```diff
@@ -1359,6 +1348,10 @@ func (sp *shardProcessor) receivedMetaBlock(headerHandler data.HeaderHandler, me
 		return
 	}
 
+	if !sp.blockTracker.ShouldAddHeader(metaBlock) {
```

Comment:
You can remove this check at all. The same situation as in metablock.go above

Reply:
done